app/models/AddressJourneyTTLModels.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import play.api.libs.json._
+import reactivemongo.bson.BSONDateTime
+import reactivemongo.play.json._
+
+case class AddressJourneyTTLModel(nino: String, editedAddress: EditedAddress) {
+  def createIndex: String =
+    s"$nino-${editedAddress.getIndex}"
+}
+
+sealed trait EditedAddress {
+  val expireAt: BSONDateTime
+  def getIndex: String
+}
+
+case class EditSoleAddress(expireAt: BSONDateTime) extends EditedAddress {
+  override def getIndex: String = EditedAddress.editSoleAddress
+}
+case class EditPrimaryAddress(expireAt: BSONDateTime) extends EditedAddress {
+  override def getIndex: String = EditedAddress.editPrimaryAddress
+}
+case class EditCorrespondenceAddress(expireAt: BSONDateTime) extends EditedAddress {
+  override def getIndex: String = EditedAddress.editCorrespondenceAddress
+}
+
+object EditedAddress {
+
+  val editSoleAddress: String = "EditSoleAddress"
+  val editPrimaryAddress: String = "EditPrimaryAddress"
+  val editCorrespondenceAddress: String = "EditCorrespondenceAddress"
+
+  val addressType = "addressType"
+  val expireAt = "expireAt"
+
+  implicit val writes = new OWrites[EditedAddress] {
+    def writes(model: EditedAddress): JsObject = Json.obj(
+      addressType -> model.getClass.getSimpleName,
```

Comment:
this is britle in that the writes refers to the model name and the reads refers to a variable, change it so they both refer to the variable name
```suggestion
      addressType -> model match {
        case EditedSoleAddress(_) => editSoleAddress
        ...
      },
```